jinahub/encoders/image/TimmImageEncoder/timm_encoder.py

```diff
@@ -0,0 +1,116 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Optional, List, Dict, Iterable, Tuple
+
+import numpy as np
+
+import torchvision.transforms as T
+import torch
+
+from jina import Executor, requests, DocumentArray
+from jina_commons.batching import get_docs_batch_generator
+
+from timm import create_model, list_models, list_modules
+from timm.data import resolve_data_config
+from timm.data.transforms_factory import create_transform
+
+
+class TimmImageEncoder(Executor):
+    """
+    :class:`TimmImageEncoder` encodes ``Document`` blobs of type `ndarray` (`float32`) and shape
+    `B x H x W x C` into `ndarray` of `B x D`.
+    Where `B` is the batch size and `D` is the Dimension of the embedding.
+    If `use_default_preprocessing=False`, the expected input shape is `B x C x H x W` with `float32` dtype.
+
+    Internally, :class:`TimmImageEncoder` wraps the pre-trained models from
+    `Timm library`.
+    https://rwightman.github.io/pytorch-image-models/
+
+    :param model_name: the name of the model. Models listed on:
+        https://rwightman.github.io/pytorch-image-models/models/
+    :param pooled_features: Defines if the features should be pooled or not.
+    :param device: Which device the model runs on. Can be 'cpu' or 'cuda'.
+    :param default_traversal_paths: Used in the encode method an defines traversal on the received `DocumentArray`.
+    :param default_batch_size: Defines the batch size for inference on the loaded PyTorch model.
+    :param use_default_preprocessing: Should the input be preprocessed with default configuration.
+    :param args:  Additional positional arguments.
+    :param kwargs: Additional keyword arguments.
+    """
+    def __init__(
+        self,
+        model_name: str = 'resnet18',
+        pooled_features: Optional[bool] = True,
+        device: Optional[str] = None,
```

Comment:
Default should be `cpu`

Reply:
I do set it to "cpu" if device is None. If I set it to "cpu" here then I will have to make it to run on cpu even when gpu is available.

Reply:
No, the user can choose where to run it. I don't like the assumption that if GPU is available, it should run on it. This is the user's decision. Please change the default